katsdpingest/ingest_session.py

```diff
@@ -643,35 +667,45 @@ def _init_baselines(self, antenna_mask: Iterable[str]) -> None:
             raise ValueError('No baselines (bls_ordering = {}, antenna_mask = {})'.format(
                 self.cbf_attr['bls_ordering'], antenna_mask))
 
-        # Pre-compute channel masks from the RFI model
-        if self._rfi_mask_model is not None:
-            bls = self.bls_ordering.sdp_bls_ordering
-            antenna_names = set(input_name[:-1] for input_name in bls.flat)
-            antennas = [katpoint.Antenna(self.telstate['{}_observer'.format(name)])
-                        for name in antenna_names]
-            ref = antennas[0].array_reference_antenna()
-            # Turn each antenna into a location East/North/Up of the reference
-            locations = {antenna.name: np.array(ref.baseline_toward(antenna))
-                         for antenna in antennas}
-            lengths = np.array([np.linalg.norm(locations[a[:-1]] - locations[b[:-1]])
-                                for (a, b) in bls]) << u.m
+        # Pre-compute channel masks from the RFI mask and band mask models
+        with katsdpmodels.fetch.requests.Fetcher() as fetcher:
+            rfi_mask_model, band_mask_model = self._load_models(fetcher)
             cbf_spw = SpectralWindow(
                 self.cbf_attr['center_freq'], None, len(self.channel_ranges.cbf),
                 bandwidth=self.cbf_attr['bandwidth'], sideband=1)
             spw = cbf_spw.subrange(self.channel_ranges.input.start,
                                    self.channel_ranges.input.stop)
             freqs = spw.channel_freqs << u.Hz
-            # Get masks as a baseline x freq array
-            masks = self._rfi_mask_model.is_masked(freqs[np.newaxis, :], lengths[:, np.newaxis])
-            # Typically the model will only have a few threshold lengths, so
-            # most rows will be the same. Reduce it to just unique rows, with
-            # a lookup table.
-            masks, indices = np.unique(masks, axis=0, return_inverse=True)
-            self.bls_channel_mask_idx = indices.astype(np.uint32)
-            self.static_masks = masks * np.uint8(STATIC)
-        else:
-            self.bls_channel_mask_idx = np.zeros(len(self.bls_ordering.sdp_bls_ordering), np.uint32)
-            self.static_masks = np.zeros((1, len(self.channel_ranges.input)), np.uint8)
+            if rfi_mask_model is not None:
```

Comment:
Given the importance of both of these masks, how about a warning if either of them are not found ?

Reply:
There is a warning (line 202).